exec/bigmachine.go

```diff
@@ -959,6 +983,43 @@ func (w *worker) Run(ctx context.Context, req taskRunRequest, reply *taskRunRepl
 	return nil
 }
 
+func (w *worker) Discard(ctx context.Context, taskName TaskName, _ *struct{}) (err error) {
+	w.mu.Lock()
+	named := w.tasks[taskName.InvIndex]
+	w.mu.Unlock()
+	if named == nil {
+		return nil
+	}
+	task := named[taskName]
+	if task == nil {
+		return nil
+	}
+	task.Lock()
+	if task.state != TaskOk {
+		// We have no results to discard if the task is not TaskOk, as it has
+		// not completed successfully.
+		task.Unlock()
+		return nil
+	}
+	task.state = TaskRunning
+	task.Unlock()
+	for partition := 0; partition < task.NumPartition; partition++ {
+		err := w.store.Discard(ctx, taskName, partition)
+		if err != nil {
+			log.Error.Printf("error discarding %v:%d: %v", taskName, partition, err)
```

Comment:
Optional: Consider saying "warning: " instead, to avoid increasing confusion about errors users should be concerned about vs. not.

Reply:
Done.